From 7bee77f9ecdb59010b17d17d8a497ea3dc6841b9 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Tue, 21 Jun 2022 14:13:32 +0200
Subject: [PATCH] arm64: dts: st: add power syscon on stm32mp251

The Cortex-M33 management needs this node as the
PWR_CPU2D2SR register is used to determine the Cortex-M33 state.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I7430051f1ad071737ec52a8c0a100dd0ebacada7
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index afab1a22757c..d703783149a8 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2116,6 +2116,11 @@ rcc: clock-controller@44200000 {
 			access-controllers = <&rifsc 156>;
 		};
 
+		pwr: syscon@44210000 {
+			compatible = "st,stm32mp25-pwr", "syscon";
+			reg = <0x44210000 0x0400>;
+		};
+
 		exti1: interrupt-controller@44220000 {
 			compatible = "st,stm32mp1-exti";
 			interrupt-controller;
@@ -2489,6 +2494,7 @@ m33_rproc: m33@0 {
 			resets = <&scmi_reset RST_SCMI_C2_R>,
 				 <&scmi_reset RST_SCMI_C2_HOLDBOOT_R>;
 			reset-names = "mcu_rst", "hold_boot";
+			st,syscfg-cm-state = <&pwr 0x204 0x0000000c>;
 			interrupt-parent = <&intc>;
 			interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
 
-- 
2.39.5

